### ISSUE TITLE
Track Toxic Exposure events in GA

### ISSUE THREAD

**Issue by pacerwow** [department-of-veterans-affairs/va.gov-team#90136]
## Issue Description
As a Product Manager I would like to track the opt-in rates of the Toxic Exposure section of the 526 form so that I can understand what percentage of users go through that conditional form flow.

Want to know how many veterans are selecting at least one condition on the page.

---
## Tasks
- [ ] Enable the analytics property on the checkbox component on the Toxic Exposure Conditions page

## Acceptance Criteria
- [ ] The dataLayer.event is visible in Google Analytics
- [ ] We can demonstrate the count of times this checkbox has been selected and not selected

### Background
We want to better understand how Veterans engage with the 526 form so that we can make informed product and design changes based on our knowledge of its use. When we have ways to identify how Veterans interact with the form, we'll be able to analyze how else Veterans use the form. E.g. Are homeless Veterans more or less likely to claim Toxic Exposure? How long does it typically take for Veterans to complete a 526 when claim PTSD? Which parts of the form take longer to complete? Why? How can we improve their experience. Additionally, quantitative data supports ongoing research opportunities by highlighting areas of friction in the UX, and supplements qualitative user research studies so that the insights are deeper and more informative.

There's a primary goal here to track the opt-in rates for Toxic Exposure, but this is also a discovery opportunity for how we might do this in the future for other form components. The scope is intentionally small here as there's an immediate need and we just want to learn how we can bring form interactions into GA4.

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by pacerwow** [comment]
Future goal (not in scope for TE 1.0)
- [ ] On the same page, be able to count the number of conditions a Veteran has selected as related to Toxic Exposure. We do not need to know the actual conditions themselves, only the count.

**Comment by pacerwow** [comment]
Here are a couple of helpful links:
[VA Component library analytics setup](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/site-wide/component-library-analytics-setup.js)

[Slack thread](https://dsva.slack.com/archives/C1Q4294JJ/p1722645773875799) I started with the analytics team to learn more about implementation. I was using Housing situation at the time as this example because it existed.

[Checkbox group](https://design.va.gov/storybook/?path=/docs/uswds-va-checkbox-group--docs) info in Storybook.

Not directly related, but demonstrates how Carbs uses custom events in UA
https://docs.google.com/spreadsheets/d/1ok7YLNdaDSFs4XG7b4IEUkwhy0EfaUYsbP5xgTRqr18/edit?gid=2087351809#gid=2087351809

[GA4 migration FAQ](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/ga4-migration-faqs#GA4MigrationFAQs-EventTracking)

**Comment by christinec-fftc** [comment]
Notes from sync with Evelyn
- for 1.1, it will be important for us to know how many are clicking on the TE link on the Review page
- another important metric would be to understand if we get any abandonment from TE

**Comment by christinec-fftc** [comment]
### Summary of findings

**Step 1** - I created a funnel exploration on staging and prod called ‘526 Toxic Exposure Opt In’. This report can tell us:
Given Veterans who started a 526 form, how many claimed at least one condition related to toxic exposure, (i.e. opted in), and from there how many completed the toxic exposure section of the form, essentially a view into the digital version of questions 15A to 15E of the pdf. If there is any concern of abandonment within the TE section, we can enhance this report by adding more of the TE pages in between.

Here is [an example](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK1ZWQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--983ff4638e4bc604d7693a6bc42777dea1c1cf65/data.pdf) from staging for the last 30 days. Since this data is from staging and we only have a few users enabled for TE, we can see that 39% of test users that start the form are filing out TE, but they all end up completing the TE flow. There is zero abandonment from our testing 🙂. This will be interesting to see with real data in prod once we start the release.

**Step 2** - Next I looked into enabling analytics for the checkbox components on the toxic-exposure/conditions page, curious as to what additional reporting this leads to.

It looks like we would need to go through a security review before we can enable the collection of this data (see [Sensitive form components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/design-system-components#DesignSystemComponents-SensitiveFormComponents). I have not started the SR process yet since the above reporting may suffice for now.

I still moved forward with this locally just to see what more we could get. I was able to enable analytics for the checkbox group and it fired two events. One with event name `int-checkbox-group-option-click` and another with `undefined` values. Here’s a sample of what the events would look like when selecting a checkbox. It appears the same events with the same values are firing when unchecking the box, so this would only let us know how many times checkboxes are being clicked and not opt-in vs opt-out.
<img width="1632" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK2hWQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--63cd76250ecf35e348461a9dc17b4e24145e94fb/image.png" alt="image.png" />

Also, this unfortunately did not enable analytics for the checkboxes themselves. Looks like this would require a small platform change, but I did it on my local to see what this could look like. Now we can see an event `int-checkbox-option-click` is fired with a value of `checkbox-checked: true`
<img width="1422" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK3RWQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--87f3eeac33402681e719b9c43582212514ede254/image.png" alt="image.png" />

**Step 3** - From here I looked into how we could incorporate this into the funnel exploration. In terms of funnel steps, GA4 seems to only let you configure to check if an event was fired and not filter the funnel based on data that was included in a given event.
<img width="619" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK3hWQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5245fbd4a7cf301944098bff8673009f91c9c651/image.png" alt="image.png" />

I searched through platform docs, slack and GA4, but at this point we are in need of help. I created [this platform support ticket](https://dsva.slack.com/archives/CBU0KDSB1/p1723491570835139), where I was instructed to create a [Google Analytics Implementation Request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=johnny-jesensky-adhoc&labels=analytics-insights%2C+analytics-request%2C+collab-cycle-review%2C+collaboration-cycle%2C+gtm&projects=&template=analytics-google-analytics-implementation-request-template.md&title=Analytics+Implementation+or+QA+Support+for+%5BTeam+Name+-+Feature+Name%5D)

**Step 4** - The last thing I looked into was whether we could fire custom events based on our criteria of opt in vs opt out so it could easily fit into the funnel exploration tooling. I dug into the web component pattern, checkbox group and checkbox web components, validation, and submit code. There is not currently a way to utilize `recordEvent()` for the conditions page without rewriting it from scratch to not use a web component pattern. If we were to rewrite the component to manage its own state and onSubmit event, then we can control calling recordEvent for our desired scenarios (1. Clicking submit button with one or more conditions selected or 2. Clicking submit button with the “I’m not claiming…” checkbox selected.) However, as with any move to a custom component instead of a platform/library component there are tradeoffs. We'd get the flexibility to build what we need, but then add to the maintenance pile for 526 and no longer get any free platform/component library improvements.

**Step 5** - The last last thing, for 1.1, adding a custom event for the 'Answer our new questions' on the alert on the Review page is trivial. We can file a new ticket for that work. Wondering if that should be its own separate funnel exploration report.

In summary, the reports created in step 1 should answer the basic questions of how many Veterans are opting into the TE flow and how many are completing it. We should discuss as a team how much further we should go at this time.

**Comment by christinec-fftc** [comment]
next steps
- step 5 is not needed. there may already be a way to do this without adding a custom even
- Jared to create GA Implementation Request for further discovery

**Comment by pacerwow** [comment]
Let's keep this open until we get to a resolution with the Platform team. I'll open an analytics implementation request ticket and see if that team can assist us with GA events.
